src/shell/shell.js

```diff
@@ -337,6 +338,139 @@ define(function(require) {
   };
 
   /**
+   * Moves the file or directory at the `source` path to the 
+   * `destination` path by relinking the source to the destination 
+   * path.
+   */
+  Shell.prototype.mv = function(source, destination, callback) {
+    var fs = this.fs;
+    var shell = this;
+    
+    callback = callback || function() {};
+
+    if(!source) {
+      callback(new Errors.EINVAL('missing source path argument'));
+      return;
+    }
+    else if(source === Constants.ROOT_DIRECTORY_NAME) {
+      callback(new Errors.EINVAL('the root directory is not a valid source argument'));
+      return;
+    }
+
+    if(!destination) {
+      callback(new Errors.EINVAL('missing destination path argument'));
+      return;
+    }
+
+    function move(sourcepath, destpath, callback) {
+      sourcepath = Path.resolve(this.cwd, sourcepath);
+      destpath = Path.resolve(this.cwd, destpath);
+      var destdir = Path.dirname(destpath);
+
+      // Recursively create any directories on the destination path which do not exist
+      shell.mkdirp(destdir, function(error) {
```

Comment:
The `mv` utility won't ensure the path is correct by creating folders. Instead, it errors if the destination directory doesn't exist.